x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/get_rule_execution_events_route.ts

```diff
@@ -6,42 +6,64 @@
  */
 
 import { transformError } from '@kbn/securitysolution-es-utils';
+import { GetAggregateRuleExecutionEventsResponse } from '../../../../../common/detection_engine/schemas/response';
 import { buildRouteValidation } from '../../../../utils/build_validation/route_validation';
 import { buildSiemResponse } from '../utils';
 import type { SecuritySolutionPluginRouter } from '../../../../types';
 
 import { DETECTION_ENGINE_RULE_EXECUTION_EVENTS_URL } from '../../../../../common/constants';
-import { GetRuleExecutionEventsRequestParams } from '../../../../../common/detection_engine/schemas/request/get_rule_execution_events_request';
-import { GetRuleExecutionEventsResponse } from '../../../../../common/detection_engine/schemas/response/get_rule_execution_events_response';
+import {
+  GetRuleExecutionEventsQueryParams,
+  GetRuleExecutionEventsRequestParams,
+} from '../../../../../common/detection_engine/schemas/request/get_rule_execution_events_schema';
 
 /**
- * Returns execution events of a given rule (e.g. status changes) from Event Log.
- * Accepts rule's saved object ID (`rule.id`).
- *
- * NOTE: This endpoint is under construction. It will be extended and finalized.
- * https://github.com/elastic/kibana/issues/119598
+ * Returns execution events of a given rule (aggregated by executionId) from Event Log.
+ * Accepts rule's saved object ID (`rule.id`), `start`, `end` and `filters` query params.
  */
 export const getRuleExecutionEventsRoute = (router: SecuritySolutionPluginRouter) => {
   router.get(
     {
       path: DETECTION_ENGINE_RULE_EXECUTION_EVENTS_URL,
       validate: {
         params: buildRouteValidation(GetRuleExecutionEventsRequestParams),
+        query: buildRouteValidation(GetRuleExecutionEventsQueryParams),
       },
       options: {
         tags: ['access:securitySolution'],
       },
     },
     async (context, request, response) => {
       const { ruleId } = request.params;
+      const {
+        start,
+        end,
+        query_text: queryText = '',
+        status_filters: statusFilters = '',
+        page,
+        per_page: perPage,
+        sort_field: sortField = 'timestamp',
+        sort_order: sortOrder = 'desc',
```

Comment:
Would prefer to set defaults on Decode (and keep route validation cleaner), but was fighting io-ts, will revisit...

Reply:
Validation is now happening on route decode as of 53cc6bf